api/products.thrift

```diff
@@ -0,0 +1,78 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+
+include "shared.thrift"
+
+namespace py ProductManagement
+namespace js ProductManagement
+
+
+/*
+struct PrivilegeRecord {
+  1: string   name,
+  2: bool     isGroup
+}
+typedef list<PrivilegeRecord> PrivilegeRecords
+*/
+
+struct DatabaseConnection {
+  1:          string engine,         // The database engine, such as "sqlite" or "postgresql".
+  2:          string host,
+  3:          i32    port,
+  4:          string username_b64,
+  5: optional string password_b64,   // Database password is NOT sent server->client!
+  6:          string database
+}
+
+/* ProductConfiguration carries administrative data regarding product settings. */
+struct ProductConfiguration {
```

Comment:
We will, yes. Product will be extended with extra data, and Product carries information for _users_.

ProductConfiguration on the other hand carries configuration data, such as database stuff, which is applicable for the - upcoming - Web admin and command line management tools.